db/migrate/20170410185513_create_checkouts.rb

```diff
@@ -0,0 +1,9 @@
+class CreateCheckouts < ActiveRecord::Migration[5.0]
+  def change
+    create_table :checkouts do |t|
+      t.timestamps null: false
+      t.references :book, index: true, foreign_key: true, null: false
+      t.references :user, index: true, foreign_key: true, null: false
```

Comment:
How do we ensure that a book can't be checked out twice? 

The UI tries to prevent this by not showing a checkout link for unavailable books, but if two users load that page at the same time, they'll both see a book as available and be able to check it out. We should be able to prevent this at the database and model level with validations and constraints/indexes.